include/boost/url/string.hpp

```diff
@@ -55,38 +104,117 @@ class string_value : public string_view
         char*& dest);
 
 public:
+    /** The default allocator type to construct read-only instances of @ref string_value
+
+        An allocator of this class can be instantiated to create
+        multiple read-only string values.
+
+        @note This custom allocator is not an instance of the C++ Allocator named requirement
+
+        @par References
+        @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator">Allocator (cppreference)</a>
+
+     */
     class allocator;
 
+    /** Destructor
+
+        Upon destruction, ownership of the shared string is released. If
+        there are no more owners, the resources for the string are released.
+
+     */
     inline
     ~string_value();
 
+    /** Default constructor
+
+        This function creates the default base @ref string_view, which points to
+        no string / nullptr.
+
+        @note As an optimization, the reference counter is `nullptr`.
```

Comment:
"@note As an optimization, the reference counter is `nullptr`." could be a comment in the implementation but it doesn't go in a javadoc